src/patches/parts/hotkeys.cc

```diff
@@ -396,20 +394,20 @@ void ExecuteSpaceAction(FleetBarViewController* fleet_bar, bool (*GetKeyDownInt)
       if (pre_scan_widget
           && (pre_scan_widget->_visibilityController->_state == VisibilityState::Visible
               || pre_scan_widget->_visibilityController->_state == VisibilityState::Show)) {
-        did_pre_scan = true;
```

Comment:
There was a reason for this at some point, I forget what it was though

Reply:
Possibly because you were not returning on each action, which the new code does